src/Jobs/IndexJob.php

```diff
@@ -32,15 +36,29 @@ public function __construct(
         ?int $batchSize = null,
         bool $processDependencies = true
     ) {
+        $this->documents = $documents;
```

Comment:
We need to add `private` fields to `IndexJob` for `documents`, `method`, `batchSize` and `processDependencies`?

Reply:
I don't believe so. Or at least, I've only ever used "magic" properties when creating Jobs in our projects.

Reply:
Please define these fields. Dynamic properties are being deprecated in PHP 8.2

https://wiki.php.net/rfc/deprecate_dynamic_properties